source/agora/network/Manager.d

```diff
@@ -535,11 +538,21 @@ public class NetworkManager
             &this.onRegisterName, Periodic.Yes);
     }
 
+    /// Called when validator set is changed
+    public void onValidatorSetChanged (ref UTXO[Hash] validator_utxos) @trusted
+    {
+        this.last_known_validator_utxos = validator_utxos;
+
+        foreach (val; this.validators())
+            if (val.identity.utxo !in this.last_known_validator_utxos)
+                if (this.peer_list.linearRemoveElement(val))
+                    this.peer_list_version++;
```

Comment:
Doesn't look like this is closing the connection though, just removing it from the list ?

Reply:
updated to call `shutdown()` on the client